controllers/cluster_controller.go

```diff
@@ -1307,6 +1307,17 @@ func checkCoordinatorValidity(cluster *fdbtypes.FoundationDBCluster, status *fdb
 			continue
 		}
 
+		processGroupStatus := processGroups[process.Locality["instance_id"]]
+		pendingRemoval := processGroupStatus != nil && processGroupStatus.Remove
+		if processGroupStatus != nil && processGroupStatus.GetConditionTime(fdbtypes.PodPending) != nil {
```

Comment:
I wonder if we should factor this into a more general method for detecting if a process group is failing in a way that means we should skip it in checks like this. Maybe this would use the same kind of logic as we use for automatic replacements?